scripts/mod_loader/altered/missions.lua

```diff
@@ -160,134 +132,27 @@ function BuildIsBoardBusyPredicate(board)
 	end
 end
 
-function Mission:MissionEnd()
-	self:MissionEndImpl()
-	
-	local fx = SkillEffect()
-	modApi:fireMissionEndHooks(self, fx)
-	fx:AddScript([[
-		modApi.runLaterQueue = {}
-		
-		modApi:conditionalHook(
-			BuildIsBoardBusyPredicate(modApi.current_mission.Board),
-			function()
-				-- BoardBusyPredicate defined above will yield true once we exit to main menu,
-				-- but when that happens, current_mission is reset to nil.
-				if modApi.current_mission then
-					modApi.current_mission.Board = nil
-					modApi.current_mission = nil
-				end
-			end
-		)
-	]])
-	Board:AddEffect(fx)
-end
-
-function Mission:SetupDifficulty()
-	-- Can be used to setup the difficulty of this mission
-	-- with regard to value returned by GetDifficulty()
-end
-
-local oldBaseStart = Mission.BaseStart
-function Mission:BaseStart(suppressHooks)
-	suppressHooks = suppressHooks or false
-
-	if not suppressHooks then
-		modApi:firePreMissionAvailableHooks(self)
-	end
-
-	Board.isMission = true
-	self.Board = Board
-	self.QueuedSpawns = {}
-
-	-- begin oldBaseStart
-	-- Copy the code rather than calling the original function, since
-	-- we want to insert a new SetupDifficulty() function before
-	-- environment start.
-	self.VoiceEvents = {}
-	
-	if self.AssetId ~= "" then
-		self.AssetLoc = Board:AddUniqueBuilding(_G[self.AssetId].Image)
-	end
-	
-	self.LiveEnvironment = _G[self.Environment]:new()
-	self:SetupDifficulty()
-
-	self.LiveEnvironment:Start()
-	self:StartMission()
-	
-	self:SetupDiffMod()
-	
-	self:SpawnPawns(self:GetStartingPawns())
-	-- end oldBaseStart
-
-	-- Clear QueuedSpawns, since the Vek burrow out immediately when entering the mission
-	self.QueuedSpawns = {}
-
-	if not suppressHooks then
-		modApi:firePostMissionAvailableHooks(self)
-	end
-
-	self.Initialized = true
-end
-
-local modLoaderHooksFired = false
-local oldApplyEnvironmentEffect = Mission.ApplyEnvironmentEffect
-function Mission:ApplyEnvironmentEffect()
-	if not modLoaderHooksFired then
-		modApi:firePreEnvironmentHooks(self)
-	end
-	
-	-- ApplyEnvironmentEffect() is supposed to return true once
-	-- it's done applying its effects, but the game's own environments
-	-- don't follow this rule...
-	local isDone = false
-	if self.LiveEnvironment:IsEffect() then
-		isDone = oldApplyEnvironmentEffect(self)
-	end
-	
-	if not modLoaderHooksFired then
-		-- Schedule the post hooks to fire once the board is no longer busy
-		modApi:conditionalHook(
-			BuildIsBoardBusyPredicate(Board),
-			function()
-				modLoaderHooksFired = false
-				if not Game or not GAME or not Board then
-					return
-				end
-				
-				modApi:firePostEnvironmentHooks(self)
-			end
-		)
-		
-		modLoaderHooksFired = true
-	end
-	
-	return isDone
-end
-
-function Mission:IsEnvironmentEffect()
-	return true
-end
+-- Why have this function been changed to always return true?
```

Comment:
It allows for environment hooks to work in all missions (as far as I understand; this has been in the mod loader before I took over maintenance)